src/inference/src/ie_core.cpp

```diff
@@ -150,27 +150,64 @@ class CoreImpl : public ie::ICore, public std::enable_shared_from_this<ie::ICore
             auto it = config.find(CONFIG_KEY(CACHE_DIR));
             if (it != config.end()) {
                 std::lock_guard<std::mutex> lock(_cacheConfigMutex);
-                _cacheConfig._cacheDir = it->second;
-                if (!it->second.empty()) {
-                    FileUtils::createDirectoryRecursive(it->second);
-                    _cacheConfig._cacheManager = std::make_shared<ie::FileStorageCacheManager>(std::move(it->second));
-                } else {
-                    _cacheConfig._cacheManager = nullptr;
+                fillConfig(_cacheConfig, it->second);
+                for (auto& deviceCfg : _cacheConfigPerDevice) {
+                    fillConfig(deviceCfg.second, it->second);
                 }
-
                 config.erase(it);
             }
         }
 
+        void setCacheForDevice(const std::string& dir, const std::string& name) {
+            std::lock_guard<std::mutex> lock(_cacheConfigMutex);
+            fillConfig(_cacheConfigPerDevice[name], dir);
+        }
+
         // Creating thread-safe copy of config including shared_ptr to ICacheManager
-        CacheConfig getCacheConfig() const {
+        // Passing empty or not-existing name will return global cache config
+        CacheConfig getCacheConfigForDevice(const std::string& name,
+                                            bool deviceSupportsCacheDir,
```

Comment:
can we have a reference to `ICore` in `CoreConfig` and in this case it will automatically detect the value of `deviceSupportsCacheDir` ?

Reply:
In this case we should pass 'Core' object with access to it's 'plugins' mutable private member. And to access it, we'll need to lock appropriate mutex of 'Core' object. And to do this, we'll need to 'unlock' this mutex before calling of 'getCacheConfigForDevice' which is very inconvenient. My suggestion is to keep it as is due to these reasons